### DISCUSSION TITLE
Can facets be used to display separate graphs?

### DISCUSSION THREAD

**Discussion by skybrian** [observablehq/plot#874] [Q&A] [answered]
I have a CSV file with two columns for a time series, with the row number implying the time.

I can easily plot them on the same graph using marks, but I think these might be best be displayed as separate graphs, one above the other. I can render them in separate cells in Observable, but facets look like they might work too? However, I don't see how to do it. (I've looked at the examples and the API, but I don't really grok facets and don't understand how they work.) How would you get all the data in a mark to be displayed in its own facet?

**Comment by Fil** [comment, marked as answer]
Facets represent a partition of your dataset—something you can’t do with this representation. However you can  transform the data so that it has one observation per row, then partition these observations: https://observablehq.com/@fil/untidy-data-874

**Reply by skybrian** [reply]
Okay, that should work. Thanks!

But I'm not too sure about these "tidy" principles. They aren't as clear-cut as they pretend. In this case, whether these columns are one "observation" or two seems like a matter of perspective and choice of visualization? They are readings from two different sensors, but they are measuring the same wheel rotation. When you take the sine and cosine of the same angle, they are not independent data. The idea is to do sensor fusion after the data is cleaned up.

**Reply by Fil** [reply]
Thanks for the context. What you're after is definitely _not_ a partition of your data, and faceting is not the correct approach. You can certainly do two charts, and even share some of their scales, but that would be “small multiples”. It takes a bit more of javascript to prepare the layout: I've added an example at the bottom of the notebook.

**Reply by Fil** [reply]
I've added yet another possibility at the end of the notebook, showing how it's possible (though admittedly not simple) to use the facet names as "facet data", and derive the values with a custom transform.

**Reply by skybrian** [reply]
Thanks for your help!
